apps/src/blockly/addons/cdoGenerator.ts

```diff
@@ -21,20 +21,25 @@ export default function initializeGenerator(
 
   // This function was a custom addition in CDO Blockly, so we need to add it here
   // so that our code generation logic still works with Google Blockly
-  blocklyWrapper.Generator.blockSpaceToCode = function (name, opt_typeFilter) {
-    const generator = blocklyWrapper.getGenerator();
-    generator.init(blocklyWrapper.mainBlockSpace);
-    let blocksToGenerate = blocklyWrapper.mainBlockSpace.getTopBlocks(
+  blocklyWrapper.Generator.blockSpaceToCode = function (name) {
+    const blocksToGenerate = blocklyWrapper.mainBlockSpace.getTopBlocks(
       true /* ordered */
     );
-    if (opt_typeFilter) {
-      if (typeof opt_typeFilter === 'string') {
-        opt_typeFilter = [opt_typeFilter];
-      }
-      blocksToGenerate = blocksToGenerate.filter(block =>
-        (opt_typeFilter as string[]).includes(block.type)
+    return blocklyWrapper.Generator.blocksToCode(name, blocksToGenerate);
+  };
+
+  // Used to generate code for an array of top blocks.
+  blocklyWrapper.Generator.blocksToCode = function (
+    name: string,
+    blocksToGenerate: Block[]
+  ) {
+    if (name !== 'JavaScript') {
```

Comment:
is `name` a cdo blockly parameter that we need to support still? It doesn't seem like it's used beyond this check.

Reply:
Every call to `blocksToCode` in the main repo does pass `'JavaScript'` in as the first argument.

In theory, we'd use it to specify which generator use (JavaScript, Python, etc.). We've only ever supported JavaScript with Blockly, but presumably that could change?

I wasn't planning to change the function's signature here though since it needs to keep working as is for CDO Blockly.